# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/nsIDragService.idl
# Commit: 384b025f56d3
# Full Hash: 384b025f56d3f87e045ecede6b1a16b71c0b5d4f
# Author: David P <daparks@mozilla.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 21 - Separate nsIDragService and nsIDragSession implementations r=gstoll,geckoview-reviewers,rkraesig,win-reviewers,m_kato
#   
#   Split the class inheritance trees for nsIDragService and nsIDragSession.
#   Remember that, before the start of this patch series, the inheritance
#   diagram was:
# ==============================================================================

diff -r beeda5ef3712 -r 384b025f56d3 widget/nsIDragService.idl
--- a/widget/nsIDragService.idl	Thu Jul 04 07:48:11 2024 +0000
+++ b/widget/nsIDragService.idl	Thu Jul 04 07:48:11 2024 +0000
@@ -128,15 +128,7 @@
     */
   nsIDragSession getCurrentSession([optional] in nsISupports aWidgetProvider);
 
-  /**
-   * TODO: Temporary proxy for old C++ callers, for whom [optional] is
-   * insufficient.  This method is removed at the end of this patch series.
-   */
 %{ C++
-  nsresult GetCurrentSession(nsIDragSession** aSession) {
-    return GetCurrentSession(nullptr, aSession);
-  }
-
   nsIDragSession* GetCurrentSession(nsIWidget* aWidget) {
   	nsCOMPtr<nsIDragSession> session;
     GetCurrentSession(aWidget, getter_AddRefs(session));
@@ -196,6 +188,20 @@
   nsIMockDragServiceController getMockDragController();
 
   /**
+   * True if this is a mock nsIDragService, created with
+   * createMockDragController().
+   */
+  readonly attribute boolean isMockService;
+
+%{ C++
+  bool IsMockService() {
+    bool ret = false;
+    MOZ_ALWAYS_SUCCEEDS(GetIsMockService(&ret));
+    return ret;
+  }
+%}
+
+  /**
    * If this is true, mSessionIsSynthesizedForTests should not become true.
    * This hack is used to bypass the "old" drag-drop test behavior, which needed
    * special behavior to pass.  The new framework requires the actual browser